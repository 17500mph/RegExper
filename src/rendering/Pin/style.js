import {
  brown,
  strokeBase
} from 'rendering/style';

export const pin = {
  fill: brown,
  ...strokeBase
};
