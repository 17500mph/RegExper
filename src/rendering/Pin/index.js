import React from 'react';

import * as style from './style';

const radius = 5;

const Pin = () => <circle
  style={ style.pin }
  r={ radius }
  transform={ `translate(${ radius } ${ radius })` }/>;

const layout = data => {
  data.box = {
    width: radius * 2,
    height: radius * 2
  };
  return data;
};

export default Pin;
export { layout };
