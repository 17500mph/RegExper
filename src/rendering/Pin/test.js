import React from 'react';
import { shallow } from 'enzyme';

import Pin, { layout } from 'rendering/Pin';

describe('Pin', () => {
  test('rendering', () => {
    const component = shallow(
      <Pin/>
    );
    expect(component).toMatchSnapshot();
  });

  test('layout', () => {
    const processed = layout({
      type: 'Pin'
    });
    expect(processed).toMatchSnapshot();
  });
});
