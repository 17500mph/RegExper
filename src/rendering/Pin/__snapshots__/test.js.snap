// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pin layout 1`] = `
Object {
  "box": Object {
    "height": 10,
    "width": 10,
  },
  "type": "Pin",
}
`;

exports[`Pin rendering 1`] = `
<circle
  r={5}
  style={
    Object {
      "fill": "#6b6659",
      "stroke": "#000",
      "strokeWidth": "2px",
    }
  }
  transform="translate(5 5)"
/>
`;
